kubetest2-gke/deployer/build.go

```diff
@@ -47,3 +52,31 @@ func (d *deployer) verifyBuildFlags() error {
 	d.BuildOptions.RepoRoot = d.RepoRoot
 	return d.BuildOptions.Validate()
 }
+
+// ensure that the version is a valid gke version
+func normalizeVersion(version string) (string, error) {
+	// (1.18.10)(-gke)(.601.123)(+existingSuffix)
+	// capture repeated occurrences of group 3 and don't capture just last
+	re, err := regexp.Compile(`(\d\.\d+\.\d*)(-\w+)?((?:\.\d+)*)(\+.*)*`)
+	if err != nil {
+		return "", err
+	}
+	matches := re.FindStringSubmatch(version)
+	if matches == nil || len(matches) != 5 {
+		return "", fmt.Errorf("%q is not a valid gke version", version)
+	}
+	fmt.Println(matches)
+	// get the core version as is
+	finalVersion := matches[1]
+	// force append -gke, ignore alpha,beta
+	finalVersion += "-gke"
+	// add the optional patch number or .0
+	if matches[3] == "" {
```

Comment:
I started using [named regex](https://stackoverflow.com/questions/20750843/using-named-matches-from-go-regex) just because it was so much nicer for me to track over indexes.

Reply:
TIL, thanks. done.